src/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs

```diff
@@ -90,12 +106,37 @@ private static bool TryReadMemoryValueFromFile(string path, out ulong result)
             return false;
         }
 
+        /// <summary>Find the cgroup memory limit path.</summary>
+        /// <returns>The limit path if found; otherwise, null.</returns>
+        private static string FindCGroupMemoryLimitPath()
+        {
+            string cgroupMemoryPath = FindCGroupPath("memory", out CGroupVersion version);
+            if (cgroupMemoryPath == null)
+            {
+                return null;
+            }
+
+            if (version == CGroupVersion.CGroup1)
+            {
+                return cgroupMemoryPath + "/memory.limit_in_bytes";
+            }
+            else if (version == CGroupVersion.CGroup2)
+            {
+                // 'memory.high' is a soft limit; the process may get throttled
+                // 'memory.max' is where OOM killer kicks in
+                return cgroupMemoryPath + "/memory.max";
+            }
+
+            return null;
+        }
+
         /// <summary>Find the cgroup path for the specified subsystem.</summary>
         /// <param name="subsystem">The subsystem, e.g. "memory".</param>
         /// <returns>The cgroup path if found; otherwise, null.</returns>
-        private static string FindCGroupPath(string subsystem)
+        private static string FindCGroupPath(string subsystem, out CGroupVersion version)
         {
-            if (TryFindHierarchyMount(subsystem, out string hierarchyRoot, out string hierarchyMount) &&
+            version = CGroupVersion.None;
+            if (TryFindHierarchyMount(subsystem, out version, out string hierarchyRoot, out string hierarchyMount) &&
                 TryFindCGroupPathForSubsystem(subsystem, out string cgroupPathRelativeToMount))
             {
                 // For a host cgroup, we need to append the relative path.
```

Comment:
I don't have a container runtime that currently supports cgroup v2 so I can't verify this host vs container difference in cgroup v2.